extra/lemon.c

```diff
@@ -2785,6 +2785,7 @@ void Parse(struct lemon *gp)
   filebuf = (char *)malloc( filesize+1 );
   if( filesize>100000000 || filebuf==0 ){
     ErrorMsg(ps.filename,0,"Input file too large.");
+    free(filebuf);
```

Comment:
An alternative option was to 1) use `xmalloc()`, 2) check file size before the allocation. But I am fine with this solution too.

Reply:
Thanks for the review! I think `xmalloc()` needed a new header and we needed `free()` anyway in case `malloc()` failed, so I decided to use `free()` in both cases.